TSRM/tsrm_win32.c

```diff
@@ -656,8 +656,12 @@ TSRM_API int shmget(key_t key, size_t size, int flags)
 	}
 	shm->segment = shm_handle;
 	shm->descriptor = MapViewOfFileEx(shm->segment, FILE_MAP_ALL_ACCESS, 0, 0, 0, NULL);
+	if (shm->descriptor == NULL) {
+		SET_ERRNO_FROM_WIN32_CODE(GetLastError());
```

Comment:
This is missing a `CloseHandle(shm->segment);`.

Reply:
Oh, right! Thank you!